libs/ml/include/ml/dataloaders/word2vec_loaders/sgns_w2v_dataloader.hpp

```diff
@@ -17,8 +17,10 @@
 //
 //------------------------------------------------------------------------------
 
+#include "math/tensor/tensor.hpp"
```

Comment:
you can just forward declare tensor instead of including it in all of these header files. Then include in the source file.